modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateFlowLocalServiceImpl.java

```diff
@@ -113,7 +90,53 @@ public ObjectStateFlow addObjectStateFlow(
 			long userId, long objectFieldId, List<ObjectState> objectStates)
 		throws PortalException {
 
-		return null;
+		ObjectStateFlow objectStateFlow = _addObjectStateFlow(
+			userId, objectFieldId);
+
+		long objectStateFlowId = objectStateFlow.getObjectStateFlowId();
+
+		List<ObjectState> newObjectStates = TransformUtil.transform(
+			objectStates,
+			objectState -> {
+				ObjectState newObjectState =
+					_objectStateLocalService.addObjectState(
+						userId, objectState.getListTypeEntryId(),
+						objectStateFlowId);
+
+				newObjectState.setObjectStateTransitions(
+					objectState.getObjectStateTransitions());
+
+				return newObjectState;
+			});
+
+		for (ObjectState sourceObjectState : newObjectStates) {
```

Comment:
@murilostodolni instead of doing two `fors` for `objectStates`/`newObjectStates` (`TransformUtil.transform` has a for inside it), can we merge both into one?

```
for (ObjectState objectState : objectStates) {
	ObjectState sourceObjectState =
		_objectStateLocalService.addObjectState(
			userId, objectState.getListTypeEntryId(),
			objectStateFlowId);

	for (ObjectStateTransition objectStateTransition :
			objectState.getObjectStateTransitions()) {

		if (Validator.isNull(
				objectStateTransition.
					getTargetObjectStateListTypeEntryId())) {

			continue;
		}

		ObjectState targetObjectState =
			_objectStateLocalService.getObjectStateFlowObjectState(
				objectStateTransition.
					getTargetObjectStateListTypeEntryId(),
				objectStateFlowId);

		_objectStateTransitionLocalService.addObjectStateTransition(
			userId, objectStateFlowId,
			sourceObjectState.getObjectStateId(),
			targetObjectState.getObjectStateId());
	}
}
```